### PR TITLE
Add support for new BlobDB status codes and TryLater retry support

### PR DESCRIPTION
**Pull request by Hexxeh**

### CODE REVIEW

**Comment by Katharine** [comment]
This will retry ~0.05 seconds later (if the queue happens to be empty; at least _something_ will be sent 0.05 seconds later), which is probably not the desired effect. It's probably preferable to actually wait a bit.

The lazy option is to completely ignore the `TryLater` response; we'll try again after the standard timeout interval. Alternatively, we'd probably want to add a "don't send before" mechanism to the queue.

**Comment by Hexxeh** [comment]
Hmm, in my testing over PULSE it certainly took way more than that, a couple seconds or so.

The default timeout of 5 seconds only allows for one (possibly two?) attempts within the SyncWrapper timeout of 10 seconds, so we should possibly either decrease the retry timeout or increase the sync timeout if we ignored the response. Any preference?

**Comment by Katharine** [comment]
The retry timeout was set on François' advice, so it'd probably be better to increase the sync timeout — probably globally, since we're below that layer at this point.

File: libpebble2/services/blobdb.py [hunk 1]
```diff
@@ -121,15 +121,20 @@ def _send_queued_data(self):
...
 
     def _handle_response(self, packet):
         with self._lock:
             if packet.token in self._pending_ack:
                 pending = self._pending_ack[packet.token]
                 del self._pending_ack[packet.token]
-                if callable(pending.callback):
+
+                if packet.response == BlobStatus.TryLater:
+                    # Do nothing, wait for the packet to timeout and re-send
```
Comment (Katharine) [inline libpebble2/services/blobdb.py]: This won't timeout; "do nothing" means leaving it in the `_pending_ack` list.

**Pull request merged**